common/src/main/java/org/apache/falcon/util/ResourceManager.java

```diff
@@ -0,0 +1,490 @@
+            } else {
+                String str = String.format("Resource Release unable to find zNode for consumerId [%s]", consumerId);
+                LOG.error(str);
+                throw new IOException(str);
+            }
+        } finally {
+            disconnect();
+        }
+    }
+
+    private static String getConsumerIdFromZnode(String data) {
+        return data.split(":")[0];
+    }
+
+    private static STATUS getConsumerStatusFromZnode(String data) {
+        return STATUS.fromVal(data.split(":")[1].charAt(0));
+    }
+
+    private static byte[] makeZNodeData(String consumerId, STATUS status) {
+        return String.format("%s:%s", consumerId, status) .getBytes();
+    }
+
+    private static String makeConsumerZNodeName(String basePath) {
+        return String.format("%s/Consumer-", basePath);
+    }
+
+
+    /**
+     *  Queue that holds the znodes in sorted order.
+     *  Znodes are sorted by status first and then by sequential number.
+     */
+
+    class ResourceQueue {
+
+        class QueueElement {
+            private String child;
+            private String childPath;
+            private STATUS status;
+            private String consumerId;
+            private int version;
+
+            public QueueElement(String consumerId, String child, String childPath, STATUS status, int version) {
+                this.consumerId = consumerId;
+                this.child = child;
+                this.status = status;
+                this.version = version;
+                this.childPath = childPath;
+            }
+        }
+
+        private SortedSet<QueueElement> q;
+
+        public ResourceQueue() {
+            q = new TreeSet<>(new Comparator<QueueElement>() {
+                @Override
+                public int compare(QueueElement o1, QueueElement o2) {
+                    String consumer1 = String.format("%s:%s", o1.status, o1.child);
+                    String consumer2 = String.format("%s:%s", o2.status, o2.child);
+                    return consumer1.compareTo(consumer2);
+                }
+            });
+        }
+
+        public void add(String child, String childPath, String zkNodeData, int version) {
+            String consumerId = getConsumerIdFromZnode(zkNodeData);
+            STATUS consumerStatus = getConsumerStatusFromZnode(zkNodeData);
+            q.add(new QueueElement(consumerId, child, childPath, consumerStatus, version));
+        }
+
+        public int findPosition(String zkNodePath) {
+            int pos = 0;
+            Iterator<ResourceQueue.QueueElement> itr = q.iterator();
+            while (itr.hasNext()) {
+                if (itr.next().childPath.equals(zkNodePath)) {
+                    break;
+                }
+                pos++;
+            }
+            return pos;
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        String zkHost = "c6402.ambari.apache.org";
+        String resourceName = "mysql-ds";
```

Comment:
please remove hard coding and read it through properties file for both the variables.